File: databinding/src/main/java/com/example/anvil/databinding/MainLayout.java
package com.example.anvil.databinding;

import android.content.Context;

import static trikita.anvil.DSL.*;

import trikita.anvil.RenderableView;

public class MainLayout extends RenderableView {

    public MainLayout(Context c) {
        super(c);
    }

    @Override
    public void onAttachedToWindow() {
        super.onAttachedToWindow();
    }

    @Override
    public void onDetachedFromWindow() {
        super.onAttachedToWindow();
    }

    @Override
    public void view() {
        frameLayout(() -> {
            size(FILL, FILL);
            textView(() -> {
                text("Hello, Anvil!");
            });
        });
    }
}

File: databinding-kotlin/src/main/kotlin/com/example/anvil/databinding/MainLayout.kt
package com.example.anvil.databinding

import android.content.Context

import trikita.anvil.DSL.*

import trikita.anvil.RenderableView

class MainLayout(c: Context) : RenderableView(c) {

    override fun onAttachedToWindow() {
        super.onAttachedToWindow()
    }

    override fun onDetachedFromWindow() {
        super.onAttachedToWindow()
    }

    override fun view() {
        frameLayout {
            size(FILL, FILL)
            textView { text("Hello, Anvil!") }
        }
    }
}
